### DISCUSSION TITLE
[v0.1500.1.0] Is medical application too fast?

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#6560] [Unstable feedback]
The ability to apply medicine directly by clicking on the "suitable treatment" button was introduced
![image](https://user-images.githubusercontent.com/73229309/132028745-6f238bf3-0527-4679-ab8f-3840a7e6f571.png)

We have 5 different ways of applying medicine now:

1. **Dragging** the syringe on body parts from inventory to medical UI
2. **Injection** by holding the syringe in your hand, holding RMB and LMB
3. **Double clicking** a medical item in inventory with the medical menu open
4. **Hotbar hotkeys** using the medical item in corresponding inventory slot
5. **Quick-healing** via suitable treatment suggestions (new)

Options 1 and 2 feel the most "physical" as they emulate holding and injecting medicine or applying bandages on specific wounds, but 3 4 and 5 are in comparison way faster.

4 in particular allows you to hit H + hotkey for almost instantaneous healing. Coupled with the upcoming medical injector, it has become very easy to apply a dose of fentanyl while being mauled by a pack of mudraptors and survive by healing on the spot.
There's also the issue of killing instantly with overdosing by spamming the shortcut button with no cooldown

Medical shortcuts have been here for a long time, and while they're convenient, I believe **the game would be better off with no quick/instant way to heal.** 

From a flavor standpoint, it's better to have doctors slam the needle into your harm (injection)

For balance purposes it should not be as fast as it currently is, as the large majority of injuries can be healed by quickly hitting the shortcuts.

In the case of dragging medicine to wounds, monsters knock you out of medical UI, and can make you drop the medicine. This natural interaction doesn't happen with hotkeys

**There's a serious gap between players who have be taught to use dragging by the tutorial and those who know about hotkey healing:**
Hotkey allows healing in combat, Dragging doesn't

My controversial opinion is that **methods 3, 4 and 5 probably shouldn't exist at all**. Yes they're convenient, but bordering on way too good. Dragging meds just works, I don't know why we need shortcuts that also break combat balance

**Comment by Rokvach** [comment]
I'd probably add a administering a medicine timer to the faster methods of injecting, so whatever it would take you to drag a medicine to a limb that amount of time would also be used when you double click, hotkey or click a suitable treatment. 
I'd also say that maybe hotkey healing would be the only thing I would add this timer to as it takes practically the same amount of time for you to double click a medicine or click the most suitable treatment than what it takes for you to drag and drop a medicine to a limb. 

I'd also classify dropping medicine when you are dragging it to a limb as a bug as picking up items from the floor is very difficult and a slow process and something we should try and avoid. Instead if you get knocked out whilst dragging a medicine that medicine should return to your inventory as the action gets cancelled.

**Comment by Ek-Videogames** [comment]
You actually forget a 6th method, the original way medical UI was 'intended' to work: select a limb highlighted and then dragging the medicine into a little healing 'receptacle' on the GUI. It was only later figured out that the "drag onto limbs" option was better because the old UI had a nasty habit of randomly deselecting the limb you wanted so you'd end up wasting bandages on the wrong limb.

On the subject of of adding an actual progression timer to injections, definitely a good idea and maybe different items should have different times, like simple medicines injecting faster, advanced meds being slower, bandages taking even slower because you have to physically apply them, etc.

Also this quote:
> This, but make it so not everything instantly kills you.
I don't give a damn either way about medical 'realism' vs 'simplicity', but the way it is now is basically reliant on the meta that literally everyone needs to have medicines on them and self-treat or at least self-stabilise to have a chance at surviving long enough for a medic to properly heal them.

> - I think make effect not instant is a good thing

> Yeah non-instant inject
but make it so we don't have to spam morphine


> - i mean even after injection, it takes a bit to actually kick in

> That's just realism semantics...
If at this 'snail-pace' of medicine we will still be able to revive people and counteract their injuries to get them stabilzed then I'd be all for it
